es/lessons/basics/documentation.md

```diff
@@ -0,0 +1,309 @@
+---
+
+Documentando código Elixir.
+
+{% include toc.html %}
+
+## Anotación
+
+Cuanto documentamos y qué hace que la documentación de calidad siga siendo un tema polémico dentro del mundo de la programación. Sin embargo, todos podemos estar deacuerdo que la documentación es importante para nosotros y para aquellos que trabajan con nuestro código.
+
+Elixir trata la documentación como *ciudadanos de primera clase*, ofreciendo varias funciones para acceder y generar documentación para tus proyectos. El núcleo de Elixir nos provee con diferentes atributos para documentar el código. Vamos a ver 3 formas:
+
+  - `#` - Para documentación en linea.
+  - `@moduledoc` - Para documentación a nivel de módulo.
+  - `@doc` - Para documentación a nivel de función.
+
+### Documentación en linea
+
+Probablemente la forma mas simple de comentar tu código es con comentarios en linea. Similar a Ruby o Python, el comentario en linea de Elixir esta denotado con `#`, frecuentemente conocido como *almohadilla* o de alguna otra forma de acuerdo de donde seas.
+
+Toma este script de Elixir (greeting.exs):
+
+```elixir
+# Outputs 'Hello, chum.' to the console.
+IO.puts "Hello, " <> "chum."
+```
+
+Elixir, cuando ejecuta este script ignorará todo desde `#` hasta el final de la linea, tratandolo como datos sin uso. Esto puede no agregar valor a la operación o rendimiento del script, sin embargo cuando no es obvio que esta pasando, un programador debería saberlo leyendo tu comentario. ¡Se consciente de no abusar del comentario en linea! Ensuciar un código puede volverse una pesadilla para algunos. Es mejor usarlo con moderación.
+
+### Documentando Módulos
+
+El anotador `@moduledoc` permite agregar documentación al nivel de módulo. Esto tipicamente se situa bajo la declaración `defmodule` al inicio del archivo. El ejemplo a continuación muestra un comentario de una linea dentro del decorador `@moduledoc`.
+
+```elixir
+defmodule Greeter do
+  @moduledoc """
+  Provides a function `hello/1` to greet a human
+  """
+
+  def hello(name) do
+    "Hello, " <> name
+  end
+end
+```
+
+Nosotros (u otros) podemos acceder a esta documentación usando la función de ayuda `h` dentro de IEx. 
+
+```elixir
+iex> c("greeter.ex")
+[Greeter]
+
+iex> h Greeter
+
+                Greeter
+
+Provides a function hello/1 to greet a human
+```
+
+### Documentando Funciones
+
+Al igual que Elixir nos da la habilidad para documentar a nivel de módulo, también habilita similares anotaciones para documentar funciones. El anotador `@doc` permite agregar documentación a nivel de función. El anotador `@doc` se situa justo sobre la función que esta comentando.
+
+```elixir
+defmodule Greeter do
+  @moduledoc """
+  ...
+  """
+
+  @doc """
+  Prints a hello message
+
+  ## Parameters
+
+    - name: String that represents the name of the person.
+
+  ## Examples
+
+      iex> Greeter.hello("Sean")
+      "Hello, Sean"
+
+      iex> Greeter.hello("pete")
+      "Hello, pete"
+
+  """
```

Comment:
This docs should be in Spanish IMHO

Reply:
Yeah, but all the code blocks are in English, by convention I guess.

Reply:
At least the comments and docs should be trandlated. Otherwise it misses the point of translating these lessons, otherwise non English speakers will have no clue what all that code is about

Reply:
Good point, @doomspork what do you think about that? this is not documented in the wiki, maybe it should be added?